daemon/src/to_sse_event.rs

```diff
@@ -272,6 +308,111 @@ fn to_cfd_state(
     }
 }
 
+fn to_cfd_details(state: model::cfd::CfdState, role: Role, network: Network) -> CfdDetails {
+    use model::cfd::CfdState::*;
+
+    let lock_label = "lock".to_string();
+    let commit_label = "commit".to_string();
+    let cet_label = "cet".to_string();
+    let refund_label = "refund".to_string();
+
+    let (txs, payout) = match state {
+        PendingOpen {
+            dlc, attestation, ..
+        }
+        | Open {
+            dlc, attestation, ..
+        } => (
+            vec![TxUrl::new(dlc.lock.0.txid(), network, lock_label)],
```

Comment:
Perhaps make 4 constructor functions that set the label to a specific value:

```rust
TxUrl::lock(....)
```

Then you can also pass the entire `dlc` in because the function knows which transaction to pull out.